src/builder.rs

```diff
@@ -483,6 +495,29 @@ impl SessionBuilder {
     }
 }
 
+/// Specifies how long the controlling ssh process should stay alive.
+#[derive(Clone, Debug, Default)]
+pub enum ControlPersist {
+    /// Will stay alive indefinitely.
+    #[default]
+    Forever,
+    /// Will be closed after the initial connection is closed
+    ClosedAfterInitialConnection,
+    /// If the ssh control server has been idle for specified duration, it will
+    /// exit.
+    IdleFor(std::time::Duration),
+}
+
+impl ControlPersist {
+    fn as_option(&self) -> String {
+        match *self {
+            ControlPersist::Forever => "ControlPersist=yes".to_string(),
+            ControlPersist::ClosedAfterInitialConnection => "ControlPersist=no".to_string(),
+            ControlPersist::IdleFor(d) => format!("ControlPersist={}s", d.as_secs()),
+        }
+    }
```

Comment:
Could use `Cow` to avoid unnecessary `String` creation:

```suggestion
    fn as_option(&self) -> String {
        match self {
            ControlPersist::Forever => Cow::Borrowed("ControlPersist=yes"),
            ControlPersist::ClosedAfterInitialConnection => Cow::Borrowed("ControlPersist=no"),
            ControlPersist::IdleFor(d) => format!("ControlPersist={}s", d.as_secs()).into(),
        }
    }
```